developer_manual/digging_deeper/rest_apis.rst

```diff
@@ -79,3 +81,66 @@ To add an additional method or header or allow less headers, simply pass additio
         }
 
     }
+
+.. _ocs-vs-rest:
+
+Relation of REST and OCS
+------------------------
+
+There is a close relationship between REST APIs and :ref:`OCS <ocscontroller>`.
+Both provide a way to transmit data between the backend of the app in the Nextcloud server and some frontend.
+
+The following combinations of attributes might be useful for various scenarios:
+
+#. Plain frontend route: No special attribute related to CSRF or CORS
+#. Plain Frontend with CRSF checks disabled: Add the ``#[NoCSRFRequired]`` attribute
+#. REST route with CORS enabled: Add the ``#[CORS]`` attribute
+#. OCS-based route
+
+There are different ways a clients might interact with your APIs.
+These ways depend on your API configuration (what you allow) and on which route the request is finally made.
+
+- *Access from web frontend* means the user is browses the Nextcloud web frontend with a browser.
+- *Access from an external app* indicates that the user is not using the normal browser (as logged in) but directly navigates a certain URL.
+  This can be in a browser tab or an external program (like an Android app or simply a curl command line).
+- *Access from external website* means that the user browses some third party web site and *magically* data from your app appears.
+  Technically, the other website would embed/load/use images, JSON data, or other resources from a URL pointing to the Nextcloud server.
+
+.. list-table:: Comparision of different API types
+    :header-rows: 1
+    :align: center
+
+    * - Description
+      - 1 (plain)
+      - 2 (no CSRF)
+      - 3 (CORS)
+      - 4 (OCS)
+    * - URL prefix (relative to server)
+      - ``/apps/<appid>/``
+      - ``/apps/<appid>/``
+      - ``/apps/<appid>/``
+      - ``/ocs/v2.php/apps/<appid>/``
+    * - Access from web frontend
+      - yes
+      - yes (CSRF risk)
+      - yes (CSRF risk)
+      - yes
+    * - Access from external app
+      - --- (CSRF protection blocks)
+      - yes
+      - yes
+      - yes
```

Comment:
```suggestion
      - no
```
OCS is still protected against CSRF attacks using the OCS-APIRequest header or a CSRF token.

Reply:
Isn't [this line](https://github.com/nextcloud/server/blob/94c529409813e03d632662283a5cd302ef8e9781/lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php#L204) disabling the csrf checks for OCS when done with a bearer token?

Reply:
Not really, it also checks that the custom header is set (https://github.com/nextcloud/server/blob/94c529409813e03d632662283a5cd302ef8e9781/lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php#L235) which is a CSRF check.

Reply:
I think, we are arguing from different viewing angles here. Yes, the custom header is some way to enforce CSRF protection. Either a token _or_ the custom header is a way to prevent access to the endpoint.

However, when thinking of external (native) apps, this would allow to access the endpoint just by sending the header without knowing the actual token. 

PS: Having reread the last comment, I see the point in splitting _direct access in browser_ (which you would not do on OCS directly) from the _native app_ approach.